src/Dialogs/ColorPreferencesDialog.vala

```diff
@@ -0,0 +1,403 @@
+                deletable: false,
+                resizable: false,
+                title: _("Color Preferences"),
+                transient_for: parent
+            );
+        }
+
+        construct {
+            get_content_area ().add (get_colors_box ());
+
+            var close_button = new Gtk.Button.with_label (_("Close"));
+            close_button.clicked.connect (destroy);
+
+            add_action_widget (close_button, 0);
+        }
+
+        private Gtk.Widget get_colors_box () {
+
+            var colors_grid = new Gtk.Grid () {
+                column_spacing = 12,
+                row_spacing = 6,
+                margin = 9,
+                margin_top = 0,
+                halign = Gtk.Align.CENTER
+            };
+
+            var window_theme_label = new SettingsLabel (_("Window Theme"));
+            var light_theme_icon = "display-brightness-symbolic";
+            var dark_theme_icon = "weather-clear-night-symbolic";
+            var window_theme_switch = new Granite.ModeSwitch.from_icon_name (light_theme_icon, dark_theme_icon) {
+                primary_icon_tooltip_text = _("Light window theme"),
+                secondary_icon_tooltip_text = _("Dark window theme")
+            };
+            Application.settings.bind ("prefer-dark-style", window_theme_switch, "active", SettingsBindFlags.DEFAULT);
+
+            console_theme_store = new Gtk.ListStore (2, typeof (string), typeof (string));
+
+            foreach (Terminal.Theme theme in Terminal.Themes.themes) {
+                Gtk.TreeIter iter;
+                console_theme_store.insert_with_values (out iter, -1, 0, theme.name, 1, theme.palette, -1);
+            };
+
+            var console_theme_label = new SettingsLabel (_("Console Theme"));
+            console_theme_combobox = new Gtk.ComboBox.with_model (console_theme_store);
+            var console_theme_cell = new Gtk.CellRendererText ();
+            console_theme_combobox.pack_start (console_theme_cell, false);
+            console_theme_combobox.set_attributes (console_theme_cell, "text", 0);
+            console_theme_combobox.active = 0;
+            console_theme_combobox.changed.connect (combobox_selection_changed);
+
+            var black_color_label = new SettingsLabel (_("Black"));
+            var red_color_label = new SettingsLabel (_("Red"));
+            var green_color_label = new SettingsLabel (_("Green"));
+            var yellow_color_label = new SettingsLabel (_("Yellow"));
+            var blue_color_label = new SettingsLabel (_("Blue"));
+            var magenta_color_label = new SettingsLabel (_("Magenta"));
+            var cyan_color_label = new SettingsLabel (_("Cyan"));
+            var grey_color_label = new SettingsLabel (_("Grey"));
+            var background_label = new SettingsLabel (_("Background"));
+            var foreground_label = new SettingsLabel (_("Foreground"));
+            var cursor_label = new SettingsLabel (_("Cursor"));
```

Comment:
These should all be suffixed with ":" and the more common en_US spelling is "Gray"
```suggestion
            var black_color_label = new SettingsLabel (_("Black:"));
            var red_color_label = new SettingsLabel (_("Red:"));
            var green_color_label = new SettingsLabel (_("Green:"));
            var yellow_color_label = new SettingsLabel (_("Yellow:"));
            var blue_color_label = new SettingsLabel (_("Blue:"));
            var magenta_color_label = new SettingsLabel (_("Magenta:"));
            var cyan_color_label = new SettingsLabel (_("Cyan:"));
            var grey_color_label = new SettingsLabel (_("Gray:"));
            var background_label = new SettingsLabel (_("Background:"));
            var foreground_label = new SettingsLabel (_("Foreground:"));
            var cursor_label = new SettingsLabel (_("Cursor:"));
```

Reply:
When deciding on grey/grey I used the same as Tilix. But I changed it now to american spelling :)